### ISSUE TITLE
[🐞]: Beautify's Trellis texture does not render properly with Better Nether's Nether Reed and Nether Mushroom. (Fabric 1.20.1)

### ISSUE THREAD

**Issue by Zhaith-Izaliel** [MehVahdJukaar/WoodGood#691]
### Before Continuing:

- [X] Always check you are using the latest version of the mods and its dependencies
- [X] Remove mod that enhances Minecraft: Optifine, Sodium, others. The issue still persists.
- [X] If you are unsure which mod is the culprit.
Disable all of your mods and enable them 1-2 mods each time to isolate the culprit
- [X] Confirm that there is no existing issue with a similar description submitted in the list of issues.

### Version - Loader

1.20.1-FABRIC

### EveryCompat Version

everycompat-1.20-2.6.90

### Moonlight Lib Version

moonlight-1.20-2.13.25-fabric

### Issue with mods

- Better Nether <9.0.10>
- Beautify: Refabricated <1.2.0+1.20.1>

### Issue Detail

Better Nether's Nether Reed and Nether Mushroom wood type don't generate the Trellis' texture from Beautify correctly. Since trellis use the log for crafting and the nether reed and nether mushroom are not exactly a type of log with a standard texture (the first is a stem while the second is a plant akin to tall grass), it renders a partial model for the nether reed and no model at all for the nether mushroom (be it in inventory or placed).

Here is the nether reed:
![image](https://github.com/user-attachments/assets/78d8ee28-367c-4d4c-bdfa-5fccb32b8d47)

Here is the nether mushroom:
![image](https://github.com/user-attachments/assets/540870ee-3cb1-49bc-bc83-c07ed20a46ef)

Here is a default trellis for comparison (oak):
![image](https://github.com/user-attachments/assets/f0b88cb0-d8ae-4478-a14f-724b563fcf9f)

I was able to fix the issue by loading a resource pack that changes the model files under `everycomp/models/block/bd/betternether/trellis/{nether_reed,nether_mushroom}/`.  

For the nether reeds, I changed every occurences of `betternether:block/reeds_middle` to `betternether:block/nether_reed_planks` in all the JSON fies in the directory. For the nether mushroom, I changed every occurences of `betternether:block/nether_mushroom_stem_side` to `betternether:block/nether_mushroom_planks` in all the JSON files in the directory.

**Comment by Xelbayria** [comment]
i'll have to blacklist the `nether_reed`, i'm told it doesn't have block, just planks. so it won't be included.

**Comment by Zhaith-Izaliel** [comment]
Yeah considering the trouble it gives, I'm not surprised. For the nether mushroom though, do you think you'll allow a fix or blacklist it as well ?

**Comment by Xelbayria** [comment]
mushroom stuff can be fixed. I need the ID of mushroom for logs and planks if they have unique texture filenames

**Comment by Zhaith-Izaliel** [comment]
Here are the "log" and planks for the nether mushroom:

```
betternether:nether_mushroom_planks,
betternether:nether_mushroom_stem,
```

The textures are:

- [nether_mushroom_stem_side.png](https://github.com/quiqueck/BetterNether/blob/1.20.3/src/main/resources/assets/betternether/textures/block/nether_mushroom_stem_side.png)
- [nether_mushroom_stem_top.png](https://github.com/quiqueck/BetterNether/blob/1.20.3/src/main/resources/assets/betternether/textures/block/nether_mushroom_stem_top.png)
- [nether_mushroom_planks.png](https://github.com/quiqueck/BetterNether/blob/1.20.3/src/main/resources/assets/betternether/textures/block/nether_mushroom_planks.png)

N.B: The "stem" corresponds to the log for the nether mushroom.

# Edit

Fixed the texture list to contain only the relevant textures.

**Comment by Xelbayria** [comment]
Just checking, the blocks using `nether_mushroom_planks` are also the wrong texture?

**Comment by Zhaith-Izaliel** [comment]
You mean the generated blocks using the planks texture ? Or the texture that I linked ?

**Comment by Zhaith-Izaliel** [comment]
The other generated assets that **don't** use the mushroom stem in their recipe works correctly (the planks texture is applied correctly), see:
![image](https://github.com/user-attachments/assets/ec4ab38d-1dce-4b41-8491-c9e060415047)

**Comment by Xelbayria** [comment]
alright, just nether_mushroom not using the side texture (it's because it has "side" in the filename) 

> You mean the generated blocks using the planks texture ? Or the texture that I linked ?

the first part.

**Comment by Zhaith-Izaliel** [comment]
> the first part.

See my above comment then :wink:

**Comment by Xelbayria** [comment]
dang. bad news: i'll have to blacklist nether_mushroom, too. it can be used for other blocks' texture that are generated based on the log's texture. 

BUT using the nether_mushroom's side_texture is not going to work.

**Comment by Xelbayria** [comment]
## FIXED in v2.6.91

**Comment by Zhaith-Izaliel** [comment]
> dang. bad news: i'll have to blacklist nether_mushroom, too. it can be used for other blocks' texture that are generated based on the log's texture.
> 
> BUT using the nether_mushroom's side_texture is not going to work.

Alright, it is fair considering the way it is made. Thanks for the fix ! I'll close the issue as resolved
